sklearn/pipeline.py

```diff
@@ -221,22 +227,58 @@ def _validate_steps(self):
                             "'%s' (type %s) doesn't"
                             % (estimator, type(estimator)))
 
+        return deepcopy(self._steps)
```

Comment:
I just found that if you set an estimator to `None`, it will raise an error when trying to clone it/this.

```
TypeError: Cannot clone object 'None' (type <type 'NoneType'>): it does not seem to be a scikit-learn estimator as it does not implement a 'get_params' methods.
```
so probably this should be fine:

```python
return [(name, clone(est)) if est is not None else (name, None)                         |    X = iris.data                                                                                
                for name, est in self._steps]
```